src/folding/hypernova/nimfs.rs

```diff
@@ -185,17 +190,18 @@ impl<C: CurveGroup> NIMFS<C> {
         }
 
         // Step 1: Get some challenges
-        let gamma: C::ScalarField = transcript.get_and_append_challenge(b"gamma").unwrap();
-        let beta: Vec<C::ScalarField> = transcript
-            .get_and_append_challenge_vectors(b"beta", ccs.s)
-            .unwrap();
+        let gamma_scalar = C::ScalarField::from_le_bytes_mod_order(b"gamma");
+        let beta_scalar = C::ScalarField::from_le_bytes_mod_order(b"beta");
+        transcript.absorb(&gamma_scalar);
```

Comment:
This would mean adding such a method to the `Transcript` trait, leaving to the user the ability to define how those bytes should be consumed by the underlying hash function? This is indeed something I thought of! 

My concern for adding this is that it could result in an overly complex api, since transcripts "theoretically" consume/return field elements only. That would add an `absorb` method not receiving a field element in its signature, leading to asking whether we should do the same wrt other types. (cc @arnaucube)

Reply:
I think that is related to what's being discussed here: https://github.com/privacy-scaling-explorations/folding-schemes/pull/40#discussion_r1421320213